activation/poet.go

```diff
@@ -212,6 +208,25 @@ func (c *HTTPPoetClient) CertifierInfo(ctx context.Context) (*url.URL, []byte, e
 	return url, certifierInfo.Pubkey, nil
 }
 
+func (c *HTTPPoetClient) verifyPhaseShiftConfiguration(ctx context.Context) error {
+	if c.fetchedPhaseShift != 0 {
+		return nil
+	}
+
+	resp, err := c.info(ctx)
+	if err != nil {
+		return err
+	} else if resp.PhaseShift.AsDuration() != c.expectedPhaseShift {
+		c.logger.Panic("verifying poet service configuration",
+			zap.Duration("server: phase shift", resp.PhaseShift.AsDuration()),
+			zap.Duration("config: phase shift", c.expectedPhaseShift),
+		)
+	}
+
+	c.fetchedPhaseShift = resp.PhaseShift.AsDuration()
```

Comment:
The `fetchPhaseShift` cannot be modified without synchronization because the HTTP client is used from multiple goroutines. It must be thread-safe. IMO it's best to avoid any kind of state in it.

Besides, I think a better place for this check is in the `poetService`. The HTTP client should only do HTTP requests.

Reply:
Yes, but we are checking phase_shift by client creation on client layer + if poet didn't answered in that moment, we have to check it again before submitting challenge. 

So that's why I think, that is better to keep  `fetchedPhaseShift` in state on client layer (and it is easy to protect it by atomic, forgot about it om Friday evening). Otherwise we have to check it every time before submitting challenge and if phase_shift didn't match with config, we lost a lot of time.

Reply:
>Yes, but we are checking phase_shift by client creation on client layer + if poet didn't answered in that moment, we have to check it again before submitting challenge.

It could be checked in `poetService` creating as well.

Doing it in the `poetService` allows for having this logic in a single place (regardless of `PoetClient` implementation) and makes testing it easier, although it requires more changes as the `PoetClient` interface would need to be changed.

@fasmat thoughts?